flink-java/src/main/java/org/apache/flink/api/java/operators/PartitionOperator.java

```diff
@@ -148,9 +147,6 @@ else if (pKeys instanceof Keys.SelectorFunctionKeys) {
 			}
 			
 		} 
-		else if (pMethod == PartitionMethod.RANGE) {
```

Comment:
Do you mean that if `pKeys.getKeyType().isSortKeyType() == false` and it's range partitioner, we should throw a exception?
If yes, I think it should be ok, for the range partitioner, it does not require the range to be sorted in meaningful way, it just try to make each target partition have roughly same size data. As long as we can compare key fields, we should be able to use range partitioner.